src/sortStudents.ts

```diff
@@ -1,16 +1,65 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const avgGrades = (students: Student): number => (
+  students.grades.reduce((acc, grade) => acc + grade, 0)
+  / students.grades.length);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const sortedStudents = [...students];
+
+  sortedStudents.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name:
+        return order === 'asc'
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name);
+
+      case SortType.Surname:
+        return order === 'asc'
+          ? a.surname.localeCompare(b.surname)
+          : b.surname.localeCompare(a.surname);
+
+      case SortType.Age:
+        return order === 'asc'
+          ? a.age - b.age
+          : b.age - a.age;
+
+      case SortType.Married:
+        return order === 'asc'
+          ? Number(a.married) - Number(b.married)
+          : Number(b.married) - Number(a.married);
+
+      case SortType.AverageGrade:
+        // eslint-disable-next-line no-case-declarations
+        const avgGradeA = avgGrades(a);
+        // eslint-disable-next-line no-case-declarations
+        const avgGradeB = avgGrades(b);
+
+        return order === 'asc'
+          ? avgGradeA - avgGradeB
+          : avgGradeB - avgGradeA;
```

Comment:
```suggestion
          ? avgGrades(a) - avgGrades(b)
          : avgGrades(b) - avgGrades(a);
```
and remove those vars